src/import/crd.ts

```diff
@@ -177,8 +150,45 @@ export class ImportCustomResourceDefinition extends ImportBase {
   }
 }
 
-function assert(condition: boolean, message: string) {
-  if (!condition) {
-    throw new Error(`invalid CustomResourceDefinition manifest: ${message}`);
+export function safeParseCrds(manifest: string): any[] {
+  const schemaPath = path.join(__dirname, '..', 'schemas', 'crd.schema.json');
+  const schema = JSON.parse(fs.readFileSync(schemaPath, { encoding: 'utf8' }));
+  const reviver = new SafeReviver({
+    sanitizers: { description: SafeReviver.DESCRIPTION_SANITIZER },
+  });
+
+  // first parse and strip
+  const objects = safeParseYaml(manifest, reviver);
+
+  // since the manifest can contain non crds as well, we first
+  // collect all crds and only apply a schema validatio on them.
+
+  const crds: any[] = [];
+
+  function collectCRDs(objs: any[]) {
+    for (const obj of objs.filter(o => o)) {
+      if (obj.kind === CRD_KIND) {
```

Comment:
nit: Would be nice if we could make this more type safe by using `objs: unknown[]`

Reply:
Not sure I see the value. What type assertion do you see missing here? We aren't using `obj.kind` because actually checking its value.